### PR TITLE
feat: standardized --version flag for create_cli consumers (Fixes #48)

### LINKED ISSUE #48: 1.0: Standardize a --version flag for create_cli() consumers
## Goal

Give every clickwork-built CLI a consistent, opt-in or default `--version` flag so consumers do not all reimplement it differently.

## Context

`clickwork.__version__` is exposed in `src/clickwork/__init__.py`, but `create_cli()` does not surface a `--version` flag for consuming CLIs. Each consumer currently has to add their own Click `--version` callback.

Once 1.0 ships, whatever pattern consumers settle on becomes the de-facto standard. Standardizing now avoids fragmentation and lets the framework own the formatting (including, if desired, both the consumer CLI version and the underlying clickwork version).

## Suggested scope

Pick one approach:

- Add a `version=` parameter to `create_cli()` and install a `--version` global option automatically when provided.
- Offer a `clickwork.add_version_option(cli, version)` helper using the same pattern as `add_global_option()`.
- Document the recommended manual pattern as the supported answer.

If implemented:

- Decide whether to also print the underlying `clickwork` version next to the consumer version.
- Add tests covering `--version` output and exit code.

## Acceptance criteria

- The README and GUIDE describe the canonical way to add `--version` to a clickwork CLI.
- If implemented as code, behavior is covered by tests and listed in the public API surface.
- The choice is consistent with the SemVer / public API policy from #36.

### PR DESCRIPTION
**Pull request by qubitrenegade**
Two new keyword-only kwargs on `create_cli()`:

- `version: str | None = None` — used verbatim.
- `package_name: str | None = None` — resolves via `importlib.metadata.version(package_name)` when `version` is `None`.

When both are `None` (current default), no `--version` flag is added. When either is set, the group gets `click.version_option(version, '-V', '--version', prog_name=name)`.

**Precedence:** explicit `version=` wins over `package_name=`.

## Error handling

`importlib.metadata.PackageNotFoundError` is caught at `create_cli()` time and re-raised as `ValueError` naming the failing distribution. Chose eager validation over Click's built-in lazy resolution so typos (`click_work` vs `clickwork`) fail at construction instead of silently hiding until a user runs `--version` in production.

## Test plan

- [x] 6 tests: literal string, `-V` short alias, auto-resolve via `package_name='clickwork'`, explicit-wins precedence, no-flag default, bogus-package ValueError
- [x] `mise exec -- python -m pytest tests/ -q` → 263 passed

Roadmap: Wave 2a #48.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/clickwork/cli.py [hunk 1]
```diff
@@ -211,6 +214,11 @@ def create_cli(
       --env           (string -- select config environment)
       --yes / -y      (flag -- skip confirmation prompts)
 
+    If ``version`` or ``package_name`` is provided, the CLI also gains
+    ``-V`` / ``--version`` at the root level (wired via
+    :func:`click.version_option`). When neither kwarg is set, no
+    version flag is installed -- existing callers see no change.
```
[the remark this answers is not in the document]
Comment (qubitrenegade) [reply src/clickwork/cli.py:220]: Fixed in 073f7ee. Updated GUIDE.md (Quick start prose + Global flags table + new 'Version flag' subsection describing `version=` / `package_name=` kwargs and precedence) and API_POLICY.md (Global flags bullet notes `--version`/`-V` is reserved but opt-in).

File: tests/unit/test_cli.py [hunk 2]
```diff
@@ -316,6 +316,173 @@ def check(ctx):
...
+        assert expected_version in result.output
+
+    def test_version_wins_over_package_name_when_both_set(self, tmp_path: Path):
+        """Explicit version= must override any package_name= lookup.
+
+        WHY: a caller who passes both is almost certainly doing
+        something deliberate (e.g. injecting a CI-computed version
+        string). Silently preferring the metadata lookup would surprise
+        them. The rule is simple and local: if ``version`` is truthy,
+        skip the metadata path entirely.
```
[the remark this answers is not in the document]
Comment (qubitrenegade) [reply tests/unit/test_cli.py]: Fixed in 52299bc. Rewrote the docstring to quote the exact `is not None` check and call out the edge case: `version=""` still wins over `package_name=` because the caller explicitly chose to suppress the metadata-resolved version.
